### PR TITLE
Canvas: Support selecting element by clicking on the border

### LINKED ISSUE #5325: Allow selecting element by clicking on the outer border
## Feature Description
Initially, the outer border was implemented so that it wouldn't be possible to select an element clicking on it, however, after consulting with Sam, the expected behavior is to allow selecting an element by clicking on the border as well.

Additionally, in discussion with George, we should include the border to the selection line as well.

- [ ] Include the border into selection and allow clicking on the border to select it.

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria
- The user should be able to select element by clicking on the outer border.

## Implementation Brief

The element needs to look exactly the same in the editor and in the front-end, the only thing that changes visually is the selection line including the border now, and the user being able to select the element by clicking on the border.

This can be achieved in a similar way as we're adding a padding:
- When border is added, the width and height of the element are increased accordingly, and decreased when the border is removed.
- The position is also adjusted accordingly, also considering rotation

We should additionally add some padding to the element to allow the size of the element content to stay the same.

The text size calculation needs to consider the text element content's size only (see `calculateTextHeight` and `calculateFitTextFontSize`)

To investigate: Edit modes for media and text -- probably should be as currently since only the content can be manipulated.

To investigate: option of adjusting the frame only? Probably better to keep frame and display in sync due to snapping, dragging, and generally not to complicate the readability.

To investigate: Still need to investigate how the shapes work since those have different implementation. It's possible that non-rectangular shapes will stay as they are -- the selection line is currently in the middle anyway so it's partially clickable and different from rectangular elements already now.

### PR DESCRIPTION
**Pull request by timarney**
## Context

Initially, the outer border was implemented so that it wouldn't be possible to select an element clicking on it, however, after consulting with Sam, the expected behavior is to allow selecting an element by clicking on the border as well.

## Summary
Adjusts calculations for get `box` [`getBox()`] to accommodate for border.

Without border click
https://www.youtube.com/watch?v=D40BMjkiyEg

With border click
https://www.youtube.com/watch?v=11K02cL_8eY

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #5325

### CODE REVIEW

**Comment by timarney** [comment]
Tested with text as well.
Screen Shot 2022-06-06 at 9 28 05 AM

**Comment by merapi** [comment]
Looks promising, found just two things to fix:

1. In shapes, the bounding box is in the middle of the border (more visible for big borders, I use 40):
image

2. Text resizing
Selected mode - ok:
image
Edit-mode - not sure if it's ok or not:
image
Started resizing (just 1px) - instant jump to big font:
image

File: packages/story-editor/src/components/canvas/frameElement.js [hunk 1]
```diff
@@ -80,7 +80,7 @@ const Wrapper = styled.div`
   &:focus,
   &:active {
     outline-color: ${({ theme, hasMask }) =>
-      hasMask ? 'transparent' : theme.colors.border.selection};
+    hasMask ? 'transparent' : theme.colors.border.selection};
```
Comment (barklund) [inline packages/story-editor/src/components/canvas/frameElement.js]: Hmm, you seem to have your browser misconfigured. Prettier definitely wants that extra white-space here, so it should be reverted.

**Review by barklund — CHANGES_REQUESTED** [review]
This PR is missing implementing this one line from the original PR:

> Additionally, in discussion with George, we should include the border to the selection line as well.

However, you will find that this single line has a lot of consequences. It means that when you add border, the element in essence grows and that leads to many other changes. The size and position of the element as displayed and updated in the design panel updates. The element when dragged now snaps according to its outer border shape, not inner border shape. And other elements now snap to this one using its outer border shape, not inner. Especially this snapping stuff might be a bit tricky, which is why this is a pretty huge ticket to take on. And there are probably other consequences too, that I haven't considered.

**Comment by timarney** [comment]
https://github.com/GoogleForCreators/web-stories-wp/pull/11628#pullrequestreview-999648851

☝️ Noting comment above as it outlines what areas are impacted by any PR trying to fix this issue.


Adding a note just as I dig around.

Looking at this again from `main`

Depending on the type of shape looks like the border is applied either fully to the outside i.e. for rectangles vs other shapes it's in the middle ... 

Screen Shot 2022-06-09 at 10 32 09 AM



Screen Shot 2022-06-09 at 11 08 30 AM

**Comment by timarney** [comment]
Going to close this as I don't think this approach is going to work.

**Pull request closed without merging**